### PR TITLE
Feat(script): Auto Proposal Approval Script

### PR DESCRIPTION
**Pull request by Muneeb147**
### Current Problem:
After submitting a proposal, we need to post approval votes manually from all nodes. This is a hassle when it comes to testing on testnet pods on test namespaces/clusters (eg: xnet, testnet, loadgen..).

### Solution:
An auto-approve script is written which polls on current submitted/pending approvals and approve it with `self` account. This `self` account is a validator having enough power to Vote. This script is deployed on all pods and hence parallel votes are casted.

This script is added inside `entrypoint.sh` file and is called inside `start_chain` function. It runs in background with interval of 10 seconds.

Before casting vote from an account, it checks for already casted vote and skip processing in that case.

**Follow-up:** It is right now a function inside `entrypoint.sh` file. We were having some issues in volume mount (ready only issue) on k8 pod. Will refactor it to a separate file once we clean-up entrypoint.sh in a separate task

### CODE REVIEW

File: bases/shared/scripts/auto-approve.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Check if the required input is provided
+if [ -z "$1" ]; then
+  echo "Error: Missing input argument."
+  echo "Usage: ./auto-approve.sh <ag_binary>"
+  exit 1
+fi
+
+# Time interval between polls (in seconds)
+POLL_INTERVAL=10
```
Comment (Muneeb147) [inline bases/shared/scripts/auto-approve.sh:11]: Will change it to 60.
Kept 10 for testing the change on k8.

File: bases/shared/scripts/auto-approve.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    PROPOSAL_IDS=$(echo $PROPOSALS | jq -r '.proposals[].id')
+
+    echo $PROPOSAL_IDS
+
+    if [ -n "$PROPOSAL_IDS" ]; then
+        for PROPOSAL_ID in $PROPOSAL_IDS; do
+            echo "Voting YES on proposal ID: $PROPOSAL_ID"
+
+            # Vote YES on the proposal
+            $AG_BINARY tx gov vote $PROPOSAL_ID yes \
```
[the remark this answers is not in the document]
Comment (Muneeb147) [reply bases/shared/scripts/auto-approve.sh:47]: Done:
https://github.com/Agoric/instagoric/pull/46/commits/a0c7013a4dea599fca0108d99b48bd34d39100cd

File: bases/shared/entrypoint.sh [hunk 3]
```diff
@@ -353,7 +355,12 @@ start_chain () {
         fi
         (cd /usr/src/agoric-sdk && node $extra /usr/local/bin/ag-chain-cosmos --home "$AGORIC_HOME" start --log_format=json $@  >> /state/app.log 2>&1)
     else
-        $(ag_binary) start --home="$AGORIC_HOME" --log_format=json $@ >> /state/app.log 2>&1 
+        $(ag_binary) start --home="$AGORIC_HOME" --log_format=json $@ >> /state/app.log 2>&1
+
+        if [ "$AUTO_APPROVE_PROPOSALS" = "true" ]; then
+            ./scripts/auto-approve.sh $(ag_binary) &
```
Comment (toliaqat) [inline bases/shared/entrypoint.sh:361]: This script may need to be mounted into the pod. 
Example: https://github.com/Agoric/instagoric/blob/37ca8b8b8d86fa4521b33736214dd88f255cb77e/bases/node/node.yaml#L328

Comment (Muneeb147) [reply bases/shared/entrypoint.sh:361]: Yes. Thanks for reminding. Will do while testing on k8 and then push the change.

**Comment by Muneeb147** [comment]
Tested on k8 with @usmanmani1122. Working fine.

File: bases/shared/entrypoint.sh [hunk 4]
```diff
@@ -343,8 +344,53 @@ start_helper () {
...
+                    
+
+                    if [ "$ACCOUNT_VOTE" == "VOTE_OPTION_YES" ]; then
+                        echo "Already voted YES on proposal ID: $PROPOSAL_ID"
+                        continue
+                    fi
+
+                    # Vote YES on the proposal
+                    $(ag_binary) tx gov vote $PROPOSAL_ID yes \
+                        --from=$FROM_ACCOUNT --chain-id=$CHAIN_ID --keyring-backend=test --home=$AGORIC_HOME --yes > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (Muneeb147) [reply bases/shared/entrypoint.sh:376]: Makes sense 👍
It was only needed when fetching proposals as it thrown to stderr when there is no Proposal

[the remark this answers is not in the document]
Comment (Muneeb147) [reply bases/shared/entrypoint.sh]: Done

**Review by usmanmani1122 — APPROVED** [review]

**Pull request merged**